firmware/src/wifi_task.h

```diff
@@ -0,0 +1,27 @@
+// wifi_task.h
+#ifndef WIFI_TASK_H
+#define WIFI_TASK_H
+
+#include <Arduino.h>
+#include <WiFi.h>
+
+#define WifiSSID "Haseeb"
+#define WifiPassword  "hasifast2023"
+
+
+class WifiTask
+{
+public:
+    WifiTask(const char *ssid, const char *password);
+    void begin();     // Initialize WiFi
+    String localIP(); // Retrieve the local IP address
+    bool status();    // Check if WiFi is connected
+    int channel();    // Get the current WiFi channel
+    String freq();    // Determine the frequency band
```

Comment:
comments are redundant if functions are named properly. why not call each of them: 
initializeWifi(), getLocalIp(), getStatus() etc. And freq does not need to be shortened. We're not counting bytes. We can call it getFrequency() and remove redundant comments.